### PR TITLE
Added canonical projection extent to map GET and layer GET

### PR DESCRIPTION
**Pull request by tbarsballe**
Map and Layer get requests now include an object named "extent" with the cannonical projection extent of the CRS of that map or layer. Object format is same as map bounds:

```
"extent":
    {
          "center": [0, 0], 
          "south": -90, 
          "north": 90, 
          "east": -180, 
          "west": -180
    }
```

### CODE REVIEW

**Comment by ischneider** [comment]
Since there is already a `bbox` and `extent` could be somewhat confusing, can this be changed to `projectionExtent` or similar?

I don't see a need for both `IO.bounds` methods? If I remove the `bounds(JSONObj, Envelope)` method and the associated casts, I'm not seeing any issues w/ compilation or tests.

**Comment by tbarsballe** [comment]
Most previously existing calls to IO.bounds passed in a ReferencedEnvelope, which extends `org.opengis.geometry.Envelope` and implements `com.vividsolutions.jts.geom.Envelope`.

The previous implementation of IO.bounds relies on the isNull method from `com.vividsolutions.jts.geom.Envelope`, which checks if the bbox is uninitialized or describes an empty geometry (maxx < minx). To preserve this check, an implementation of IO.bounds with `com.vividsolutions.jts.geom.Envelope` is needed.

The call to CRS.getEnvelope returns an `org.opengis.geometry.Envelope`. This seems to be a somewhat higher-level implementation than `com.vividsolutions.jts.geom.Envelope`, and doesn't include an isNull method. Consequently, the new IO.bounds method using `org.opengis.geometry.Envelope` is needed.

In summary, to preserve the existing functionality, both implementations of IO.bounds are necessary. If we want to drop the isNull check, then we could just use the new one.

As for changing `extent` to `projectionExtent`, sounds good, I will make that change.

**Comment by ischneider** [comment]
I understand the difference.

It probably depends on whether a valid center is needed or not. My guess is that the null check is simply to avoid an NPE as `getCenter` returns null if the Envelope is null.

As noted, `isNull` is a simple check so it would be easy to add this (if the invalid center causes problems)

So with the existing approach, a null envelope (regardless of type) will output:
`{"west":0.0,"south":0.0,"east":-1.0,"north":-1.0}`

If we use a singular method with no null check, the output for a null envelope will be:
{"west":0.0,"south":0.0,"east":-1.0,"north":-1.0,"center":[-0.5,-0.5]}

**Comment by tbarsballe** [comment]
Fair enough. I'll run through some tests with just the new method, and see if we get any failures. If not, then we can go with just the one method as it is now. Otherwise, I'll update the new method to match the functionality of the old one.

**Comment by tbarsballe** [comment]
Did some basic testing, seems to work fine. Updated with changes.

**Comment by eshon** [comment]
Can we just merge this?  I need to use this to update front-end projection stuff.

**Comment by tbarsballe** [comment]
I've tested the changes, and they seem safe, I'll merge.
